azure/network-security-group/examples/subnets/main.tf

```
@@ -0,0 +1,87 @@
+provider "azurerm" {
+  features {}
+}
+
+locals {
+  location = "uksouth"
+  tags = {
+    module     = "network_security_group"
+    example    = "subnets"
+    usage      = "demo"
+    department = "technical"
+    owner      = "Dee Vops"
+  }
+  resource_prefix = "tfmex-snet-nsg"
+}
+
+resource "azurerm_resource_group" "main" {
+  name     = "${local.resource_prefix}-rg"
+  location = local.location
+  tags     = local.tags
+}
+
+module "network" {
+  source  = "ans-coe/virtual-network/azurerm"
+  version = "1.3.0"
+
+  name                = "${local.resource_prefix}-vnet"
+  location            = local.location
+  resource_group_name = azurerm_resource_group.main.name
+  tags                = local.tags
+
+  address_space = ["10.0.0.0/16"]
+  subnets = {
+    snet-prod = {
+      prefix = "10.0.0.0/24"
+    }
+    snet-dev = {
+      prefix = "10.0.1.0/24"
+    }
+  }
+  include_azure_dns = true
+}
+
+module "nsg" {
+  source = "../../"
+
+  name                = "${local.resource_prefix}-nsg"
+  location            = local.location
+  resource_group_name = azurerm_resource_group.main.name
+  tags                = local.tags
+
+  subnet_ids = {
```

would it make more sense to get a **set** of ID's rather than a map. Subnet names are not globally unique so you could potentially end up with a map with duplicate names for subnets.

```suggestion
  subnet_ids = [
    module.network.subnets["snet-prod"].id,
    module.network.subnets["snet-dev"].id
  ]
```

The name and resource group of the subnet can be calculated based on the information in the ID however upon review of the code, they are not needed.

---

Tried this.  It causes issues when referencing the resource block index so I changed it to a map to make referencing easier.

---

changed to a list and used count = length(var.subnet_ids)